cmd/controller-manager/app/controller_manager.go

```diff
@@ -170,40 +176,44 @@ func Run(controllerManagerOptions *options.ControllerManagerServer) error {
 		panic("unreachable")
 	}
 
-	// TODO: leader election for this controller-manager
-	// id, err := os.Hostname()
-	// if err != nil {
-	// 	return err
-	// }
-	// rl := resourcelock.EndpointsLock{
-	// 	EndpointsMeta: v1.ObjectMeta{
-	// 		Namespace: "kube-system",
-	// 		Name:      "kube-controller-manager",
-	// 	},
-	// 	Client: leaderElectionClient,
-	// 	LockConfig: resourcelock.ResourceLockConfig{
-	// 		Identity:      id,
-	// 		EventRecorder: recorder,
-	// 	},
-	// }
+	if !controllerManagerOptions.LeaderElection.LeaderElect {
+		run(make(<-chan (struct{})))
+		panic("unreachable")
+	}
+
+	// Identity used to distinguish between multiple cloud controller manager instances
+	id, err := os.Hostname()
+	if err != nil {
+		return err
+	}
+
+	// Lock required for leader election
+	rl := resourcelock.EndpointsLock{
+		EndpointsMeta: metav1.ObjectMeta{
+			Namespace: "service-catalog",
```

Comment:
It seems like the Jenkins job failed because this namespace didn't exist. The Jenkins is running the walkthrough with the controller-manager running in the `catalog` namespace; are we assuming here that it must be running in the `service-catalog` namespace? If so, this contradicts the walkthrough readme.